fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java

```diff
@@ -347,10 +354,25 @@ public static WindowOperator standardize(AnalyticExpr analyticExpr) {
                 return sortKey;
             }
         });
-
         return sortedGroups;
     }
 
+    private static boolean isPrefixHyperPartitionSet(List<Object> hyperSet, List<Object> subSet) {
+        if (hyperSet.isEmpty() && subSet.isEmpty()) {
+            return true;
+        }
+
+        List<List<Object>> partitionPrefix = IntStream.rangeClosed(1, hyperSet.size())
+                .mapToObj(i -> hyperSet.subList(0, i)).collect(Collectors.toList());
+        return partitionPrefix.contains(subSet);
+    }
+
+    private static boolean isPrefixHyperSortSet(List<Object> hyperSet, List<Object> subSet) {
+        List<List<Object>> partitionPrefix = IntStream.rangeClosed(0, hyperSet.size())
+                .mapToObj(i -> hyperSet.subList(0, i)).collect(Collectors.toList());
+        return partitionPrefix.contains(subSet);
```

Comment:
for (int i = 0;i < subSet.size(); i++) {
            if (!hyperSet.get(i).equals(subSet.get(i))) {
                return false;
            }

            return true;
        }

Reply:
Same as above